x-pack/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -721,7 +739,7 @@ export class TaskRunner<
 
         return { interval: retryInterval };
       }),
-      monitoring,
+      monitoring: this.ruleMonitoring.getMonitoring(),
     };
   }
```

Comment:
> Should the warning field be populated with execute?

This seems to be consistent with how we're populating the last_run field

> In this scenario, I also see `nextRun` getting to set to `null`.

yea so in line 661 of `task_runner.ts`, set the `nextRun` like so:
```    
    let nextRun: string | null = null;
    if (isOk(schedule)) {
      nextRun = getNextRun({ startDate: startedAt, interval: schedule.value.interval });
    }
```
So we could unconditionally just set the `nextRun` according to the value of the `taskSchedule`, even if the rule failed

Reply:
Yea I think that makes sense because the rule is a recurring task so it will be rescheduled according to the schedule interval whether it succeeds or fails.